Assets/Scripts/Switch.cs

```diff
@@ -4,74 +4,114 @@
 
 public class Switch : MonoBehaviour
 {
+    private enum SwitchColor { Yellow, Blue, Red, Green };
+
+    [SerializeField]
+    [Tooltip("List of audio clips the switch can play." +
+        "The first is played when it turns on, the second when it turns off.")]
+    private List<AudioClip> switchClips;
+
     [SerializeField]
-    [Tooltip("The color of the switch. 0 if yellow, 1 if blue, 2 if red, 3 if green.")]
-    [Range(0, 3)]
-    private int switchColor;
+    [Tooltip("The color of the switch.")]
+    private SwitchColor switchColor;
 
     [SerializeField]
-    [Tooltip("The puzzle this switch is connected to.\nMust have a script which implements the IPuzzle interface.")]
+    [Tooltip("The puzzle this switch is connected to.\n" +
+        "Must have a script which implements the IPuzzle interface.")]
     private GameObject connectedPuzzle;
 
     [SerializeField]
-    [Tooltip("The character this switch adds to the solution.")]
+    [Tooltip("The character added to the puzzle's alphanumeric " +
+        "solution when the switch is activated.")]
     private char solutionEntry = '1';
 
+    [SerializeField]
+    [Tooltip("The name of the Animator bool for switch color.")]
+    private string animColorBool = "Color";
+
+    [SerializeField]
+    [Tooltip("The name of the Animator bool for switch on.")]
+    private string animOnBool = "On";
+
     /// <summary>
     /// The animator component attached to the switch
     /// </summary>
     private Animator animator;
 
+    /// <summary>
+    /// The current audio clip to play
+    /// </summary>
+    private AudioClip currentClip;
+
+    /// <summary>
+    /// The audio source component attached to the switch
+    /// </summary>
+    private AudioSource audioSource;
+
     /// <summary>
     /// Is the switch active or not?
+    /// The player can only use it if it is not already active.
     /// </summary>
-    private bool isSwitchOn;
+    private bool isSwitchActive_useProperty;
```

Comment:
I actually like the On name better because active already means something in Unity.

Reply:
Document somewhere that On means the switch is to the right